client/app/reducers/ManageAvailabilityReducer.js

```diff
@@ -45,25 +44,26 @@ const withChange = (state, action, day) => {
 };
 
 const ranges = (bookings) =>
-  bookings.map((b) => ({
-    start: moment(b.attributes.start),
-    end: moment(b.attributes.end),
-  }));
+  bookings.map((b) => (new Map({
+    start: moment(b.getIn([':attributes', ':start'])),
+    end: moment(b.getIn([':attributes', ':end'])),
+  })));
 
 const splitRanges = (dateRanges) =>
-  _.flatMap(dateRanges, (range) => {
-    const { start, end } = range;
+  dateRanges.flatMap((range) => {
+    const start = range.get('start');
+    const end = range.get('end');
     const days = end.diff(start, 'days');
 
-    return _.range(days).map((i) => start.clone().add(i, 'days'));
+    return new Range(0, days).map((i) => start.clone().add(i, 'days'));
   });
 
 const mergeNovelty = (state, novelty) => {
-  const blocks = splitRanges(ranges(novelty.blocks));
-  const bookings = splitRanges(ranges(novelty.bookings));
+  const blocks = splitRanges(ranges(novelty.get('blocks')));
+  const bookings = splitRanges(ranges(novelty.get('bookings')));
 
   return state.set('reservedDays', state.get('reservedDays').concat(bookings))
-       .set('blockedDays', state.get('blockedDays').concat(blocks));
+              .set('blockedDays', state.get('blockedDays').concat(blocks));
```

Comment:
Maybe this PR is the  wrong place to fix this, but just commenting here that only adding the new bookings and blocks doesn't result in correct data. It should also remove existing bookings and blocks that are not relevant anymore.